test/abilities/dry-skin.test.ts

```diff
@@ -27,131 +30,131 @@ describe("Abilities - Dry Skin", () => {
+      .ability(AbilityId.BALL_FETCH);
   });
 
-  it("during sunlight, lose 1/8 of maximum health at the end of each turn", async () => {
-    await game.classicMode.startBattle(SpeciesId.CHANDELURE);
+  it.each([
+    ["Harsh Sunlight", WeatherType.SUNNY],
+    ["Extremely Harsh Sunlight", WeatherType.HARSH_SUN],
+  ])("looses 1/8 of max health in '%s' weather, at the end of each turn", async (_name, weather) => {
+    const { override, classicMode, field, move } = game;
+    override.weather(weather);
 
-    const enemy = game.scene.getEnemyPokemon()!;
-
-    // first turn
-    game.move.select(MoveId.SUNNY_DAY);
+    await classicMode.startBattle(SpeciesId.CHANDELURE);
+    const enemy = field.getEnemyPokemon();
+    move.use(MoveId.SPLASH);
     await game.toEndOfTurn();
-    expect(enemy.hp).toBeLessThan(enemy.getMaxHp());
 
-    // second turn
-    enemy.hp = enemy.getMaxHp();
-    game.move.select(MoveId.SPLASH);
-    await game.toEndOfTurn();
-    expect(enemy.hp).toBeLessThan(enemy.getMaxHp());
+    expect(enemy).toHaveTakenDamage(toDmgValue(enemy.getMaxHp() / 8));
   });
 
-  it("during rain, gain 1/8 of maximum health at the end of each turn", async () => {
-    await game.classicMode.startBattle(SpeciesId.CHANDELURE);
-
-    const enemy = game.scene.getEnemyPokemon()!;
-
-    enemy.hp = 1;
-
-    // first turn
-    game.move.select(MoveId.RAIN_DANCE);
+  it.each([
+    ["Rain", WeatherType.RAIN],
+    ["Heavy Rain", WeatherType.HEAVY_RAIN],
+  ])("gains 1/8 of max health in '%s' weather, at the end of each turn", async (_name, weather) => {
+    const { override, classicMode, field, move } = game;
+    override.weather(weather);
+
+    await classicMode.startBattle(SpeciesId.CHANDELURE);
+    const enemy = field.getEnemyPokemon();
+    enemy.hp = enemy.getMaxHp() - toDmgValue(enemy.getMaxHp() / 8); // remove 1/8 of max health
+    move.use(MoveId.SPLASH);
     await game.toEndOfTurn();
-    expect(enemy.hp).toBeGreaterThan(1);
 
-    // second turn
-    enemy.hp = 1;
-    game.move.select(MoveId.SPLASH);
-    await game.toEndOfTurn();
-    expect(enemy.hp).toBeGreaterThan(1);
+    expect(enemy).toHaveFullHp();
   });
 
   it("opposing fire attacks do 25% more damage", async () => {
-    game.override.moveset([MoveId.FLAMETHROWER]);
-    await game.classicMode.startBattle(SpeciesId.CHANDELURE);
-
-    const enemy = game.scene.getEnemyPokemon()!;
     const initialHP = 1000;
-    enemy.hp = initialHP;
+    const { override, classicMode, field, move } = game;
+    vi.spyOn(ReceivedTypeDamageMultiplierAbAttr.prototype, "apply");
+
+    await classicMode.startBattle(SpeciesId.CHANDELURE);
+    const player = field.getPlayerPokemon();
+    // Mock the random multiplier to always return 1 (100%)
+    vi.spyOn(player, "randSeedIntRange").mockReturnValue(100);
+    const enemy = field.getEnemyPokemon();
+    // Mock the random multiplier to always return 1 (100%)
+    vi.spyOn(enemy, "randSeedIntRange").mockReturnValue(100);
```

Comment:
```suggestion
    const enemy = field.getEnemyPokemon();
```
We already mock the RNG to only do max rolls, this shouldn't be necessary.
https://github.com/Despair-Games/poketernity/blob/8eadf4c7d787ecdce6f5cbbc0f1b679379e7e389/test/test-utils/game-manager.ts#L94
(The `Pokemon#randSeedInt[Range]` methods call `BattleScene#randBattleSeedInt`)
https://github.com/Despair-Games/poketernity/blob/8eadf4c7d787ecdce6f5cbbc0f1b679379e7e389/src/field/pokemon.ts#L4398-L4400 https://github.com/Despair-Games/poketernity/blob/8eadf4c7d787ecdce6f5cbbc0f1b679379e7e389/src/field/pokemon.ts#L4388-L4390

Reply:
I didn't remember that, thanks for letting me know